README.md

```diff
@@ -1,2 +1,151 @@
+    <img alt="Git Town logo" src="https://raw.githubusercontent.com/git-town/git-town/main/website/src/logo.svg">
+  </picture>
+</p>
+
+# Git Town Action V1
+
+This action visualizes your stacked changes when proposing pull requests on GitHub:
+
+- `main`
+  - ["git town park" command #3181](https://github.com/git-town/git-town/pull/3181) :point_left:
+    - ["git town observe" command" #3186](https://github.com/git-town/git-town/pull/3186)
+
+This allows you to easily see all related PRs for a given pull request, where
+you are in the stack, as well as navigate between PRs in a stack.
+
+It is designed to work out of the box with [Git Town](https://github.com/git-town/git-town) v12+,
+but also supports previous versions via [manual configuration](#manual-configuration).
+
+## What's New
+
+Please refer to the [release page](https://github.com/git-town/action/releases/latest) for
+the latest release notes.
+
+## Getting Started
+
+### Create the GitHub Actions Workflow File
+
+Create a workflow file called `git-town.yml` under `.github/workflows` with the following
+contents:
+
+```yaml
+name: Git Town
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  git-town:
+    name: Display the branch stack
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: git-town/action@v1
+```
+
+Once this workflow is committed, the action will visualize your stacked changes
+whenever a pull request is created or updated. It also will automatically read
+your `.git-branches.toml` file to determine the main and perennial branches for
+your repository.
+
+### Modify the Pull Request Template
+
+By default, this action will append the visualization to the bottom of the PR description.
+If you are using a [pull request template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository),
+you can specify the location of the visualization in the template by adding the
+`<!-- branch-stack -->` tag to it:
+
+> Don't worry! Since the tag is a _HTML comment_, it will not be visible when viewing
+  your PR on GitHub!
+
+```md
+## Stack
+
+<!-- branch-stack -->
+
+## Checklist
+
+[ ] Foo
+[ ] Bar
+[ ] Baz
+```
+
+This action will insert the visualization directly below the tag. It will also
+leave behind the tag so that the next time this action runs, it will be able
+to use it again to update the visualization:
+
+```md
+## Stack
+
+<!-- branch-stack --> 👈 Still there!
+
+- `main`
+  - \#1 :point_left:
+    - \#2
+
+## Checklist
+
+[ ] Foo
+[ ] Bar
+[ ] Baz
+```
+
+:exclamation: Be careful not to add any content between the tag and the
+visualization, as this action will replace that content each time it
+updates your PR. Adding content above the tag, or below the list is
+safe though!
```

Comment:
Reading this documentation, does it make sense to only insert the stack visualization if there is more than one branch in it? In other words, does it make sense to not show the branch visualization if there would be only one branch in it?

Reply:
When I first introduced this action with my team, this is what it initially did. However the feedback I got was that this was confusing. People weren't sure whether there was an issue with their PR or not. This was compounded by the fact that stacked diffs was a completely different mental model for them as well. Following that feedback, I changed it so that the action always added the visualisation, and that worked out better for my team.

Admittedly, this might not be the case for other users and teams. Perhaps we could run with this for now and gather feedback from the community? I'm more than happy to add support for this (perhaps via an input), but first and foremost I'm really keen to get this in the hands of our citizens so then we can start gathering feedback.

What are your thoughts on this?

Reply:
I think if there is no stack, it could show placeholder text like:

_[not a stacked change]_


But you have more experience with how this feels when being used in real life, so I support whatever you decide here.

Reply:
Ah very true, we could definitely display a placeholder until there is an actual stack! I'd be interested to see how other tools (e.g. Graphite) do this. I'll hold off on any more changes for now to prioritise getting this merged in & released, but definitely open minded about this and down to experiment!

Reply:
Tracked in https://github.com/git-town/action/issues/4